ulib/FStar.Matrix.fst

```diff
@@ -145,87 +168,165 @@ let one_row_matrix_fold_aux #c #eq #m #n (cm:CE.cm c eq) (generator : matrix_gen
     (SB.init m (fun i -> SP.foldm_snoc cm (SB.init n (generator i))));
   matrix_seq_of_one_row_matrix generator;
   eq.symmetry rhs lhs
-             
-#push-options "--ifuel 1 --fuel 0 --z3rlimit 20" 
+
+
+let fold_of_subgen_aux #c #eq (#m:pos{m>1}) #n (cm: CE.cm c eq) (gen: matrix_generator c m n) (subgen: matrix_generator c (m-1) n) : Lemma
+  (requires subgen == (fun (i: under (m-1)) (j: under n) -> gen i j))
+  (ensures forall (i: under (m-1)). SP.foldm_snoc cm (SB.init n (subgen i)) ==
+                               SP.foldm_snoc cm (SB.init n (gen i))) =         
+  let aux_pat (i: under (m-1)) : Lemma (SP.foldm_snoc cm (SB.init n (subgen i)) 
+                                     == SP.foldm_snoc cm (SB.init n (gen i))) = 
+  SB.lemma_eq_elim (SB.init n (subgen i)) (SB.init n (gen i)) in
+  Classical.forall_intro aux_pat          
+
+let arithm_aux (m: pos{m>1}) (n: pos) : Lemma ((m-1)*n < m*n) = ()
+
+let terminal_case_aux #c #eq (#p:pos{p=1}) #n (cm:CE.cm c eq) (generator: matrix_generator c p n) (m: pos{m<=p}) : Lemma 
+  (ensures SP.foldm_snoc cm (SB.slice (seq_of_matrix (init generator)) 0 (m*n)) `eq.eq`
+           SP.foldm_snoc cm (SB.init m (fun (i:under m) -> SP.foldm_snoc cm (SB.init n (generator i)))))
+  = one_row_matrix_fold_aux cm generator
+
+#push-options "--quake 10/10 --ifuel 0 --fuel 1 --z3rlimit 10"
```

Comment:
Could you please remove the `quake` option from here since it is a debugging option.

Reply:
ok but keep in mind there's currently at least one quake in FStar.Seq.Permutation.fst

Reply:
May be get rid of that too? : )